service/resource/endpoint/update.go

```diff
@@ -3,32 +3,46 @@ package endpoint
 import (
 	"context"
 
+	"k8s.io/apimachinery/pkg/api/errors"
+
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
 )
 
-func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
-	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState interface{}) error {
+	k8sEndpoint, err := toK8sEndpoint(updateState)
 	if err != nil {
-		return nil, microerror.Mask(err)
+		return microerror.Mask(err)
 	}
 
-	update, err := r.newUpdateChange(ctx, obj, currentState, desiredState)
-	if err != nil {
-		return nil, microerror.Mask(err)
+	if k8sEndpoint == nil {
+		return nil // Nothing to do.
 	}
 
-	patch := framework.NewPatch()
-	patch.SetCreateChange(create)
-	patch.SetUpdateChange(update)
+	if isEmptyEndpoint(*k8sEndpoint) {
+		return nil
+	}
 
-	return patch, nil
-}
+	_, err = r.k8sClient.CoreV1().Endpoints(k8sEndpoint.Namespace).Update(k8sEndpoint)
+	if errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
 
-func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState interface{}) error {
 	return nil
 }
 
-func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	return nil, nil
+func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
+	k8sEndpoint, err := r.newCreateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	patch := framework.NewPatch()
+
+	patch.SetCreateChange(k8sEndpoint)
+	patch.SetUpdateChange(k8sEndpoint)
```

Comment:
Why is this still? I do not understand this. How can it be that the create state and the update state are computed the same way? I think there should be a different mechanism.

Reply:
Otherwise I would say there is no need for the create state when everything is just an update anyway.

Reply:
I did this initially with the thought that `Apply*Change` functions check for `IsAlreadyExists` (at create)  or `IsNotFound`(at update). 

I will refactor this, to have the check for existence in private methods and align with our other operators.

Reply:
Done.